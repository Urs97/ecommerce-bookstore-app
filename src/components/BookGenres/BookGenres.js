import React, { useContext } from 'react';
import './BookGenres.scss';

import StoreContext from '../../context/StoreContext';

const BookGenres = () => {
    const context = useContext(StoreContext);
    
    return (
        <section className="book-genres">
            <h4 className="sidebar-title">Book Genres</h4>
            <ul>
                <li><button onClick={() => context.changeCategory('science_fiction')}
                    className="btn-to-link">Science Fiction</button></li>
                <li><button onClick={() => context.changeCategory('history')} 
                    className="btn-to-link">History</button></li>
                <li><button onClick={() => context.changeCategory('short_stories')} 
                    className="btn-to-link">Short Stories</button></li>
                <li><button onClick={() => context.changeCategory('fantasy')} 
                    className="btn-to-link">Fantasy</button></li>
                <li><button onClick={() => context.changeCategory('biography')} 
                    className="btn-to-link">Biography</button></li>
                <li><button onClick={() => context.changeCategory('programming')} 
                    className="btn-to-link">Programming</button></li>
            </ul>
        </section>
    )
}

export default BookGenres;