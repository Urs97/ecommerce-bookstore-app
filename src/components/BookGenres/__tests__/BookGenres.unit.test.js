import React from 'react';
import {shallow} from 'enzyme';
import BookGenres from '../BookGenres';

describe('BookGenres', () => {
    test('renders BookGenres', () => {
        const wrapper = shallow(
            <BookGenres />
        );
        expect(wrapper).toMatchSnapshot();
    });
});