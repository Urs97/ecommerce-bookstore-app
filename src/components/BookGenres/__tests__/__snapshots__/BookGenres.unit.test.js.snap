// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookGenres renders BookGenres 1`] = `
<section
  className="book-genres"
>
  <h4
    className="sidebar-title"
  >
    Book Genres
  </h4>
  <ul>
    <li>
      <Link
        offset={0}
        smooth={true}
        spy={true}
        to="page-header"
      >
        <button
          className="btn-to-link"
          onClick={[Function]}
        >
          Science Fiction
        </button>
      </Link>
    </li>
    <li>
      <Link
        offset={0}
        smooth={true}
        spy={true}
        to="page-header"
      >
        <button
          className="btn-to-link"
          onClick={[Function]}
        >
          History
        </button>
      </Link>
    </li>
    <li>
      <Link
        offset={0}
        smooth={true}
        spy={true}
        to="page-header"
      >
        <button
          className="btn-to-link"
          onClick={[Function]}
        >
          Short Stories
        </button>
      </Link>
    </li>
    <li>
      <Link
        offset={0}
        smooth={true}
        spy={true}
        to="page-header"
      >
        <button
          className="btn-to-link"
          onClick={[Function]}
        >
          Fantasy
        </button>
      </Link>
    </li>
    <li>
      <Link
        offset={0}
        smooth={true}
        spy={true}
        to="page-header"
      >
        <button
          className="btn-to-link"
          onClick={[Function]}
        >
          Biography
        </button>
      </Link>
    </li>
    <li>
      <Link
        offset={0}
        smooth={true}
        spy={true}
        to="page-header"
      >
        <button
          className="btn-to-link"
          onClick={[Function]}
        >
          Programming
        </button>
      </Link>
    </li>
  </ul>
</section>
`;
